src/components/SingleArticle.jsx

```diff
@@ -35,6 +39,19 @@ const IndividualArticle = () => {
         })
     }
 
+    if (isError) {
```

Comment:
Could you refactor this to use one 404 page that takes a prop to define what went wrong? You wouldn't need new state, your catch could force navigate to a 404 page.